Feature/sync med origin

dangernil opened the pull request:
Sync with master

fredrikhgrelland committed:
Initial commit

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/afd02e54d4eda5d86d384d0205e2cf8e8646e0b4)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/9f1367fa46f61690be689739e8da41d6a0b18cde)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/19d28e673cee303c926a462a2f2920a0e089d2a8)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/19cbba14643af7c33bb034bb43feeaadf436ff6c)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/19cbba14643af7c33bb034bb43feeaadf436ff6c)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/f72fa86322b6c50b0e2b02df33e879b92a71610b)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/4de1874cf71126f6b7f628ec8511c92e02041bad)

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/62a1d8db737b865892a3a905f6d9367603e9d5d3)

fredrikhgrelland committed:
Update Vagrantfile.default

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/62a1d8db737b865892a3a905f6d9367603e9d5d3)

fredrikhgrelland committed:
Update 01_build_docker_image.yml

Commit:
Update from https://github.com/fredrikhgrelland/vagrant-hashistack/commit/cb98b9335795c7bc148afa189112c458bb9bb836)

fredrikhgrelland committed:
Merge pull request #3 from fredrikhgrelland/fredrikhgrelland-patch-1

Update 01_build_docker_image.yml

zhenik committed:
fix links in doc

zhenik committed:
Merge pull request #6 from fredrikhgrelland/fix/doc-links

Fix links in doc

zhenik committed:
feature/template-example-test (#15)

* update test command

* comment terraform-required_providers

* update cd/ci test with condition

fredrikhgrelland committed:
Update Vagrantfile.default

claesgill committed:
Added check for consul binary

claesgill committed:
Moved to pre requisites section

claesgill committed:
Upd changelog

claesgill committed:
Update CHANGELOG.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

dangernil committed:
updated the make clean to include more tmp files and folders

dangernil committed:
fixed formatting

claesgill committed:
Update Makefile

Co-authored-by: Neha Sinha <67954397+Neha-Sinha2305@users.noreply.github.com>

claesgill committed:
Added check for terraform binary

dangernil committed:
Update Makefile

Co-authored-by: Neha Sinha <67954397+Neha-Sinha2305@users.noreply.github.com>

dangernil committed:
Merge pull request #22 from fredrikhgrelland/issue_8_update_clean

updated the make clean to include more tmp files and folders

claesgill committed:
Merge pull request #20 from fredrikhgrelland/issue_9_check_for_consul_binary

Issue 9: Check for consul and terraform binary

fredrikhgrelland committed:
Update Vagrantfile.default

fredrikhgrelland committed:
Merge pull request #18 from fredrikhgrelland/fredrikhgrelland-patch-1

Update Vagrantfile.default

claesgill committed:
Added prettifyer helper command and prettifyed tf-files (#21)

pdmthorsrud committed:
Added doc on PKI (#26)

claesgill committed:
Added pre-commit helper command (#27)

zhenik committed:
Add consul namespace documentation (#30)

* add documentation

* add link for namespace management policy

dangernil committed:
fixed conflicts

claesgill reviewed (approved):
Ser bra ut for min del. Har kjørt opp og ser at postgres fungerer som forventet med `psql`.

Men om det mangler noe obvious i noen av readmene etc er jeg usikker på :sweat_smile: :shrug:

dangernil committed:
Merge branch 'master' into feature/sync_med_origin

claesgill commented on LICENSE:
```diff
@@ -198,4 +198,8 @@
\ No newline at end of file
+
+   limitations under the License.
+=======
```
```suggestion
=======
```

claesgill replied:
```suggestion
```

claesgill commented on LICENSE:
```diff
@@ -198,4 +198,8 @@
\ No newline at end of file
+
+   limitations under the License.
+=======
+   limitations under the License.
```
```suggestion
```

claesgill commented on Makefile:
```diff
@@ -50,5 +87,13 @@ clean: destroy-box remove-tmp
+pre-commit: check_for_docker_binary
+	docker run -e RUN_LOCAL=true -v "${PWD}:/tmp/lint/" github/super-linter
+	terraform fmt -recursive && echo "\e[32mTrying to prettify all .tf files.\e[0m"
+=======
```
```suggestion
```

claesgill commented on README.md:
````diff
@@ -93,5 +98,476 @@ vault kv get -field='username' secret/postgres
 vault kv get -field='password' secret/postgres
 ```
 
+=======
````
Ser det er en del `=======` i noen av filene. Ville gått gjennom å sjekket alle filene og fjernet disse.

claesgill reviewed (changes requested):
Ser det er en del `=======` i noen av filene. Ville gått gjennom å sjekket alle filene og fjernet disse. Det er sikkert lettest å gjøre med søkefunksjon i Intellij :)

zhenik commented on CHANGELOG.md:
```diff
@@ -14,3 +14,23 @@
 ### Added
 
 - Initial draft
+
+## [0.0.1 UNRELEASED]
+
+
+### Changed
+
+- updated documentation: consul namespaces #28
+- trigger tests for template_example by condition #12
+- updated the make clean in makefile to remove more tmp files/folders #8
+
+### Added
+
+- Added command that formats/prettify all `.tf`-files in directory #21 (overwritten by #24)
+- Added check for consul and terraform binary in the `Makefile` #20
+- Added `make pre-commit` command that use local linter and formatts/prettyfies all `.tf` files #24
+- `Vault PKI` section to README
+
+### Fixed
+
+- links in documentation #5
```
I think this part is taken from the original template. Check existing [CHANGELOG.md](https://github.com/fredrikhgrelland/terraform-nomad-postgres/blob/master/CHANGELOG.md)

```suggestion
```

zhenik commented on LICENSE:
```diff
@@ -198,4 +198,6 @@
    See the License for the specific language governing permissions and
-   limitations under the License.
\ No newline at end of file
+
+   limitations under the License.
+
```
```suggestion
```

zhenik commented on Makefile:
```diff
@@ -32,10 +56,23 @@ else
 destroy-box:
 	vagrant destroy -f
 
+remove-tmp:
+
```
```suggestion
```

zhenik commented on Makefile:
```diff
@@ -50,5 +87,12 @@ clean: destroy-box remove-tmp
 update-box:
 	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#if-you-are-behind-a-transparent-proxy\n\n' && exit 2)
 
+
+	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#proxy\n\n' && exit 2)
+
```
```suggestion
update-box:
	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#proxy\n\n' && exit 2)

```

zhenik commented on README.md:
```diff
@@ -1,6 +1,11 @@
+
+
+# terraform-nomad-postgres
+
```
```suggestion
```

zhenik commented on README.md:
````diff
@@ -93,5 +98,476 @@ vault kv get -field='username' secret/postgres
 vault kv get -field='password' secret/postgres
 ```
 
````
@dangernil I think the whole README.md file is wrong. It supposed to describe postgres module, not template

zhenik commented on README_template.md:
```diff
@@ -0,0 +1,51 @@
+<!-- markdownlint-disable MD041 -->
```
You need to move all content from `README_template.md` to `README.md` and vice versa

zhenik commented on dev/ansible/README.md:
```diff
@@ -3,4 +3,5 @@
 
-Se [example](../../test_example/dev/ansible)
\ No newline at end of file
+Se [example](../../test_example/dev/ansible)
```
Broken link
```suggestion
```

zhenik commented on dev/vagrant/conf/consul/README.md:
```diff
@@ -5,4 +5,6 @@ Any valid configuration added to this directory will append the configuration, i
 Adding a file `99-override.hcl` you will ensure it will be appended last, and 00-override.hcl will be read first.
 Any valid configuration from [https://www.consul.io/docs/agent/options.html#configuration_files](https://www.consul.io/docs/agent/options.html#configuration_files) will work.
+
 See [example](../../../../test_example/dev/vagrant/conf/consul/99-override.hcl)
```
```suggestion
```

zhenik commented on dev/vagrant/conf/nomad/README.md:
```diff
@@ -16,4 +16,7 @@ client {
 
-See [example](../../../../test_example/dev/vagrant/conf/nomad/99-override.hcl)
\ No newline at end of file
+
+See [example](../../../../test_example/dev/vagrant/conf/nomad/99-override.hcl)
```
Directory was renamed from `test_example` to `template_example`
```suggestion
```

zhenik commented on dev/vagrant/conf/pre_bootstrap/README.md:
```diff
@@ -12,4 +12,6 @@ The files e.g. 0-example.yml must only include pure ansible task syntax:
 
-See [example](../../../../test_example/dev/vagrant/conf/pre_bootstrap/00-prestart-example.yml)
\ No newline at end of file
+See [example](../../../../test_example/dev/vagrant/conf/pre_bootstrap/00-prestart-example.yml)
```
```suggestion
```

zhenik commented on dev/vagrant/conf/vault/README.md:
```diff
@@ -6,4 +6,5 @@ Any valid configuration added to this directory will append the configuration, i
 
-See [example](../../../../test_example/dev/vagrant/conf/vault/99-override.hcl)
\ No newline at end of file
+See [example](../../../../test_example/dev/vagrant/conf/vault/99-override.hcl)
```
```suggestion
```

dangernil committed:
Update CHANGELOG.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

dangernil committed:
Update LICENSE

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

dangernil committed:
Update README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update dev/ansible/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update dev/vagrant/conf/consul/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update dev/vagrant/conf/nomad/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update dev/vagrant/conf/pre_bootstrap/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update dev/vagrant/conf/vault/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

removed ===== from placed it did not belong

dangernil committed:
fixed github workflow

fixed github workflow

dangernil committed:
fixing linter

fixed linter

Neha-Sinha2305 commented on .github/action/create-env.py:
```diff
@@ -4,12 +4,19 @@
 
 with open(str(os.environ.get("GITHUB_WORKSPACE")) + "/" + str(os.environ.get("FILE_NAME")), "w") as text_file:
     text_file.write(out_file)
+filepath
```
```suggestion
filepath
```
We need to remove this from vagrant-hashistack code as well.

Neha-Sinha2305 commented on Vagrantfile:
```diff
@@ -10,3 +10,6 @@ Vagrant.configure("2") do |config|
         ansible.playbook = "playbook.yml" # Note this playbook is, in this context, /ansible/playbook.yml
     end
 end
+default_vagrantfile = "Vagrantfile.default"
+load default_vagrantfile if File.exists?(default_vagrantfile)
```
This file is not in order..it should just have the last two lines 13,14. Rest should be removed.
https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/Vagrantfile

dangernil committed:
removed text in README.md

dangernil committed:
made suggested changes

zhenik commented on README.md:
```diff
@@ -1,6 +1,3 @@
-<!-- markdownlint-disable MD041 -->
-<p align="center"><a href="https://github.com/fredrikhgrelland/vagrant-hashistack-template" alt="Built on"><img src="https://img.shields.io/badge/Built%20from%20template-Vagrant--hashistack--template-blue?style=for-the-badge&logo=github"/></a><p align="center"><a href="https://github.com/fredrikhgrelland/vagrant-hashistack" alt="Built on"><img src="https://img.shields.io/badge/Powered%20by%20-Vag ...
-
```
Why you have removed this part? IMHO, it should not be removed

dangernil replied:
I agree, adding it back in

zhenik commented on .github/action/create-env.py:
```diff
@@ -4,12 +4,19 @@
     if key.startswith("ENVKEY_"):
         out_file += key.split("ENVKEY_")[1] + "=" + os.environ.get(key) + "\n"
 
 with open(str(os.environ.get("GITHUB_WORKSPACE")) + "/" + str(os.environ.get("FILE_NAME")), "w") as text_file:
     text_file.write(out_file)
```
```suggestion
```

dangernil committed:
Update .github/action/create-env.py

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

dangernil committed:
added back in text which was mistakenly taken out of README.md

dangernil committed:
Merge branch 'feature/sync_med_origin' of https://github.com/fredrikhgrelland/terraform-nomad-postgres into feature/sync_med_origin

zhenik reviewed (approved):
LGTM

I would suggest to squash last ~10-12 commits
